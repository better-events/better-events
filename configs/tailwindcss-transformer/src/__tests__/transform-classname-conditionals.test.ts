import { describe, expect, it } from "vitest";
import { transform } from "../index";

describe("transform", () => {
	it("should transform className with conditionals", () => {
		const result = transform(
			`const Button: React.FC<ButtonProps> = ({ children, display }) => {
      return <button className={display === 'flex' ? 'flex' : 'inline-flex'}>{children}</button>
    }`,
			{
				styleCache: new Map(),
			},
		);
		expect(result).toMatchSnapshot();
	});
});
