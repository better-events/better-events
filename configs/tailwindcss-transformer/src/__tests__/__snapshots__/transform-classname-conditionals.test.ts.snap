// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transform > should transform className with conditionals 1`] = `
"const Button: React.FC<ButtonProps> = ({ children, display }) => {
      return <button className={display === 'flex' ? "kf-222f930b" : "kf-12330685"}>{children}</button>;
    }"
`;
